virtme/commands/run.py

```diff
@@ -1293,28 +1301,51 @@ def do_it() -> int:
     else:
         virtme_init_cmd = "virtme-init"
 
+    initsh = []
+    if args.systemd:
+        kernelargs.append("virtme_systemd=1")
+        kernelargs.append(f"console={arch.serial_console_args()[0]}")
+        initsh.extend(
+            ["mount --bind /dev/null " + unit for unit in get_conf("systemd.masks")]
+        )
+
     if args.root == "/":
-        initcmds = [f"init={guest_tools_path}/{virtme_init_cmd}"]
+        if args.systemd:
+            initsh.append(f"SYSTEMD_UNIT_PATH={CACHE_DIR}: exec /sbin/init")
+            initcmds = ["init=/bin/sh", "--", "-c", "; ".join(initsh)]
+        else:
+            initcmds = [f"init={guest_tools_path}/{virtme_init_cmd}"]
     else:
         virtfs_config = VirtFSConfig(
             path=guest_tools_path,
             mount_tag="virtme.guesttools",
         )
         export_virtfs(qemu, arch, qemuargs, virtfs_config)
-        initcmds = [
-            "init=/bin/sh",
-            "--",
-            "-c",
-            ";".join(
+        virtfs_config = VirtFSConfig(
+            path=str(CACHE_DIR),
+            mount_tag="virtme.cache",
+        )
+        export_virtfs(qemu, arch, qemuargs, virtfs_config)
```

Comment:
Is this always needed or only in case of systemd?

Reply:
Good catch, it's only for systemd (the serial-getty@.service file is written to the CACHE_DIR).

Reply:
Fixed in https://github.com/arighi/virtme-ng/pull/280/commits/918d2069374938b3936c6fbe60d8b5bda72a7aa6